### ISSUE TITLE
workflowStopped event enhancement 

### ISSUE THREAD

**Issue by uday-mandava** [argoproj/argo-workflows#7618]
# Both successful workflows and stopped workflows are generating similar kind of events with 'reason' as 'WorkflowSucceeded'. There is no way to differentiate successful vs stopped events to trigger different down streams based on the status. 

What change needs making?
Would like to see reason as 'workflowStopped' for stopped workflows.

# Use Cases
We are using [Argo events](https://argoproj.github.io/argo-events/concepts/architecture/) architecture. We are using K8s Resource eventsource,  relying on these k8 events to trigger downstreams. Since we are relying on 'WorkflowSucceeded' , it is matching events for both successful and stopped workflows.

---

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

**Comment by alexec** [comment]
Can you please clarify what you mean by "workflow stopped"? Do you mean you ran "argo stop my-wf"?

**Comment by uday-mandava** [comment]
> 

Yes, but through the API call /api/v1/workflows/{namespace}/{my-wf}/stop

**Comment by isubasinghe** [comment]
I was not able to replicate this issue. I get "workflow shutdown with strategy: Stop". 
This corresponds to https://github.com/argoproj/argo-workflows/blob/95cd467c7e75c38548bf42a5f3c940ac61568e4b/workflow/controller/workflowpod.go#L145

**Comment by alexec** [comment]
@isubasinghe @uday-mandava please make sure you do NOT use Kubernetes events for any automation, they are not reliable:

https://blog.argoproj.io/maybe-dont-use-kubernetes-events-for-automation-53951bdfd602

**Comment by AnuragThePathak** [comment]
Hello can I work on this issue?
